tests/proc/lif/test_models.py

```diff
@@ -0,0 +1,350 @@
+        lif = LIF(shape=shape,
+                  du=0, dv=0,
+                  bias=2*np.ones(shape, dtype=np.int32),
+                  bias_exp=6*np.ones(shape, dtype=np.int32),
+                  vth=8)
+        spr = VecRecvProcess(shape=(num_steps, shape[0]))
+        sps.out_ports.s_out.connect(lif.in_ports.a_in)
+        lif.out_ports.s_out.connect(spr.in_ports.s_in)
+        rcnd = RunSteps(num_steps=num_steps)
+        rcfg = LifRunConfig(select_tag='fixed_pt')
+        lif.run(condition=rcnd, run_cfg=rcfg)
+        spk_data_through_run = spr.spk_data.get()
+        lif.stop()
+        expected_spk_data = np.zeros((num_steps, shape[0]))
+        expected_spk_data[3:10:4, :] = 1
+        self.assertTrue(np.all(expected_spk_data == spk_data_through_run))
+
+    def test_bitacc_pm_impulse_du(self):
+        """
+        Send a single activation of magnitude 128 with a current decay of
+        2047 and voltage decay of 0. Current state variable decays by half every
+        time-step. Actual current values, compared to the floating point test
+        above are left-shifted by 6 bits due to bit-accurate fixed point
+        implementation with respect to Loihi hardware
+        """
+        shape = (1,)  # a single neuron
+        num_steps = 8
+        # send activation of 128. at timestep = 1
+        sps = VecSendProcess(shape=shape, num_steps=num_steps,
+                             vec_to_send=128*np.ones(shape, dtype=np.int32),
+                             send_at_times=np.array([True, False, False,
+                                                     False, False, False,
+                                                     False, False]))
+        lif = LIF(shape=shape,
+                  du=2047,  # decay_u = 1/2
+                  dv=0,
+                  bias=np.zeros(shape, dtype=np.int16),
+                  bias_exp=np.ones(shape, dtype=np.int16),
+                  vth=256*np.ones(shape, dtype=np.int32))  # high thr, no spikes
+        spr = VecRecvProcess(shape=(num_steps, shape[0]))
+        sps.out_ports.s_out.connect(lif.in_ports.a_in)
+        lif.out_ports.s_out.connect(spr.in_ports.s_in)
+        rcnd = RunSteps(num_steps=1)
+        rcfg = LifRunConfig(select_tag='fixed_pt')
+        lif_u = []
+        # Run 1 timestep at a time and collect state variable u
+        for j in range(num_steps):
+            lif.run(condition=rcnd, run_cfg=rcfg)
+            lif_u.append(lif.u.get().astype(np.int32)[0])
+        lif.stop()
+        expected_u_timeseries = [1 << (13-j) for j in range(8)]
+        expected_float_u = [1 << (7-j) for j in range(8)]
+        self.assertListEqual(expected_u_timeseries, lif_u)
+        self.assertListEqual(expected_float_u, np.right_shift(np.array(
+            lif_u), 6).tolist())
+
+    def test_bitacc_pm_impulse_dv(self):
+        """
+        Send a single activation of magnitude 128 with a voltage decay of
```

Comment:
Hopefully now you see, that it would be more worthwhile to start with WHAT this unit test is about and WHY we are doing this rather than deep diving into the nitty gritty details of the math. Again this might be more useful after the WHAT and WHY or through inline comments.
This will go over most people's head when they read this for the first time.

Reply:
Done.